pymc/distributions/multivariate.py

```diff
@@ -289,10 +293,76 @@ def logp(value, mu, cov):
         return check_parameters(
             norm - 0.5 * quaddist - logdet,
             ok,
-            msg="posdef",
+            msg="posdef covariance",
         )
 
 
+class PrecisionMvNormalRV(SymbolicRandomVariable):
+    r"""A specialized multivariate normal random variable defined in terms of precision.
+
+    This class is introduced during specialization logprob rewrites, and not meant to be used directly.
+    """
+
+    name = "precision_multivariate_normal"
+    extended_signature = "[rng],[size],(n),(n,n)->(n)"
+    _print_name = ("PrecisionMultivariateNormal", "\\operatorname{PrecisionMultivariateNormal}")
+
+    @classmethod
+    def rv_op(cls, mean, tau, *, rng=None, size=None):
+        rng = normalize_rng_param(rng)
+        size = normalize_size_param(size)
+        cov = pt.linalg.inv(tau)
+        next_rng, draws = multivariate_normal(mean, cov, size=size, rng=rng).owner.outputs
+        return cls(
+            inputs=[rng, size, mean, tau],
+            outputs=[next_rng, draws],
+        )(rng, size, mean, tau)
+
+
+@_logprob.register
+def precision_mv_normal_logp(op: PrecisionMvNormalRV, value, rng, size, mean, tau, **kwargs):
+    [value] = value
+    k = value.shape[-1]
+    delta = value - mean
+    # Equivalent, but possibly faster / more stable than slogdet, which relies on LU factorization
+    # det_sign, logdet_tau = pt.nlinalg.slogdet(tau)
+    cholesky_diagonal = pt.diagonal(pt.linalg.cholesky(tau), axis1=-2, axis2=-1)
+    logdet = 2 * pt.log(cholesky_diagonal).sum()
+
+    quadratic_form = delta.T @ tau @ delta
+    logp = -0.5 * (k * pt.log(2 * pt.pi) - logdet + quadratic_form)
+    return check_parameters(
+        logp,
+        (cholesky_diagonal > 0).all(-1),
```

Comment:
I assume this is the right check for posdef-ness? @aseyboldt

Reply:
The cholesky will simply fail (throw an exception) if the matrix is not posdef. I think we check for that in the perform method and return nan, but I don't know what for instance jax will do.

Reply:
Okay so we could check for nan? I'll see what JAX does

Reply:
I guess all if nan will also fail so this ends up being the same thing?